src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt

```diff
@@ -219,10 +219,11 @@ class UserService(
       is ClientResult.Failure -> staffUserDetailsResponse.throwException()
     }
 
-    if (deliusUser.teams?.firstOrNull()?.borough?.code != null && user.probationDeliveryUnit?.deliusCode !== deliusUser.teams.firstOrNull()?.borough?.code) {
-      when (val probationDeliveryUnit = findProbationDeliveryUnit(deliusUser.teams)) {
+    val userTeams = orderUserDeliusTeams(deliusUser)
+    if (userTeams?.firstOrNull()?.borough?.code != null && user.probationDeliveryUnit?.deliusCode !== userTeams.firstOrNull()?.borough?.code) {
```

Comment:
i don't think we want referential integrity checks provided by !==.

similar to java i suspect comparing strings by reference could leave to false negatives. I think != should be used instead.

Reply:
I know that !== is for referential comparing IO used it because i find that in some parts of the project they using it to compare nullable strings

Reply:
In that case, i'm pretty sure that code is wrong too as it could lead to false negatives but in that context the side affect is minor as all it will do is unnecessarily update the user - https://www.baeldung.com/kotlin/string-comparison.

In kotlin using == to compare things that may be null is safe (unlike java) https://kotlinlang.org/docs/equality.html#structural-equality

(on a wider note, that userHasChanged function is broken for other reasons see APS-579)

Reply:
I agree that using == to compare strings is better approach but I still not get why it used === in the project

Reply:
I suspect it was a just a mistake

Reply:
possibly by someone who mostly codes javascript, i think === is more common there possibly